src/modules/flight_mode_manager/tasks/AutoFollowTarget/FlightTaskAutoFollowTarget.cpp

```diff
@@ -64,306 +70,355 @@ bool FlightTaskAutoFollowTarget::activate(const vehicle_local_position_setpoint_
+	follow_target_status.orbit_angle_setpoint = _orbit_angle_setpoint_rad;
 
-		case FOLLOW_GIMBAL_MODE_3D:
-			// Point the gimbal at the target's 3D coordinates
-			gimbal_height = -(_position(2) - (target_position_filtered(2)));
-			break;
+	// [Debug] Log orbit angle max & actual rate setpoint
+	follow_target_status.max_angular_rate_setpoint = _orbit_angle_traj_generator.getVelSp();
+	follow_target_status.angular_rate_setpoint = _orbit_angle_traj_generator.getCurrentVelocity();
 
-		case FOLLOW_GIMBAL_MODE_2D_WITH_TERRAIN:
-			// Point the gimbal at the ground level in this tracking mode
-			gimbal_height = _dist_to_ground;
-			break;
-
-		default:
-			break;
-		}
+	// [Debug] Log desired Raw Follow position
+	const Vector3f drone_desired_position_raw = calculate_desired_drone_position(_target_position_velocity_filter.getState(), _target_course_rad + _follow_angle_rad);
+	drone_desired_position_raw.copyTo(follow_target_status.desired_position_raw);
 
-		point_gimbal_at(target_to_drone_xy.norm(), gimbal_height);
+	// Etc
+	follow_target_status.in_emergency_ascent = in_emergency_ascent;
+	follow_target_status.gimbal_pitch = gimbal_pitch;
 
-	} else {
-		// Control setpoint: Stay in current position
-		_position_setpoint(0) = _position_setpoint(1) = NAN;
-		_velocity_setpoint.xy() = 0;
-	}
-
-	// Publish status message for debugging
-	_target_position_filter.getState().copyTo(follow_target_status.pos_est_filtered);
-	follow_target_status.timestamp = hrt_absolute_time();
-	follow_target_status.emergency_ascent = _emergency_ascent;
-	follow_target_status.gimbal_pitch = _gimbal_pitch;
 	_follow_target_status_pub.publish(follow_target_status);
 
 	_constraints.want_takeoff = _checkTakeoff();
 
 	return true;
 }
 
-float FlightTaskAutoFollowTarget::update_follow_me_angle_setting(int param_nav_ft_fs) const
+float FlightTaskAutoFollowTarget::convertFollowPerspectiveToRadians(const kFollowPerspective follow_perspective) const
 {
-	switch (param_nav_ft_fs) {
-	case FOLLOW_PERSPECTIVE_BEHIND:
-		return FOLLOW_PERSPECTIVE_BEHIND_ANGLE_DEG;
-
-	case FOLLOW_PERSPECTIVE_FRONT:
-		return FOLLOW_PERSPECTIVE_FRONT_ANGLE_DEG;
-
-	case FOLLOW_PERSPECTIVE_FRONT_RIGHT:
-		return FOLLOW_PERSPECTIVE_FRONT_RIGHT_ANGLE_DEG;
-
-	case FOLLOW_PERSPECTIVE_FRONT_LEFT:
-		return FOLLOW_PERSPECTIVE_FRONT_LEFT_ANGLE_DEG;
-
-	case FOLLOW_PERSPECTIVE_MID_RIGHT:
-		return FOLLOW_PERSPECTIVE_MID_RIGHT_ANGLE_DEG;
-
-	case FOLLOW_PERSPECTIVE_MID_LEFT:
-		return FOLLOW_PERSPECTIVE_MID_LEFT_ANGLE_DEG;
-
-	case FOLLOW_PERSPECTIVE_BEHIND_RIGHT:
-		return FOLLOW_PERSPECTIVE_BEHIND_RIGHT_ANGLE_DEG;
-
-	case FOLLOW_PERSPECTIVE_BEHIND_LEFT:
-		return FOLLOW_PERSPECTIVE_BEHIND_LEFT_ANGLE_DEG;
-
-	case FOLLOW_PERSPECTIVE_MIDDLE_FOLLOW:
-		return FOLLOW_PERSPECTIVE_BEHIND_ANGLE_DEG;
-
-	default:
-		// No or invalid option
-		break;
+	if (follow_perspective < 0 || follow_perspective >= kFollowPerspectiveInvalid) {
+		return math::radians(kFollowPerspectiveAnglesDeg[kFollowPerspectiveInvalid]);
```

Comment:
eep! this will access something out of the bounds of the array. Maybe choose an existing follow perspective to default? e.g. Back? Whatever you like the best.